utility/src/main/java/org/oppia/android/util/profile/CurrentUserProfileIdIntentDecorator.kt

```diff
@@ -0,0 +1,32 @@
+package org.oppia.android.util.profile
+
+import android.content.Intent
+import org.oppia.android.app.model.ProfileId
+import org.oppia.android.util.extensions.getProtoExtra
+import org.oppia.android.util.extensions.putProtoExtra
+
+private const val PROFILE_ID_INTENT_DECORATOR = "PROFILE_ID_INTENT_DECORATOR"
+
+/** Decorator for wrapping a user's profile [ProfileId] within its intent. */
+object CurrentUserProfileIdIntentDecorator {
+  /**
+   * Packs the intent with a [ProfileId] proto object that sets it as the current
```

Comment:
```suggestion
   * Packs [this] intent with a [ProfileId] proto object that sets it as the current
```

(Since it's an extension function).

Ditto for extractCurrentUserProfileId below.

Reply:
Noted.